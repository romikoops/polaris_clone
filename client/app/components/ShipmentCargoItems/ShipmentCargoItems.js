import React, { Component } from 'react';
import { ValidatedInput } from '../ValidatedInput/ValidatedInput';
import PropTypes from 'prop-types';
import { Checkbox } from '../Checkbox/Checkbox';
import styles from './ShipmentCargoItems.scss';
import defs from '../../styles/default_classes.scss';
// import Select from 'react-select';
import { NamedSelect } from '../NamedSelect/NamedSelect';
import '../../styles/select-css-custom.css';
import { v4 } from 'node-uuid';

export class ShipmentCargoItems extends Component {
    constructor(props) {
        super(props);
        this.state = {
            cargoItemTypes: [{}],
            firstRenderInputs: !this.props.nextStageAttempt
        };
        this.handleCargoChange = this.handleCargoChange.bind(this);
        this.addNewCargo = this.addNewCargo.bind(this);
        this.toggleDangerousGoods = this.toggleDangerousGoods.bind(this);
        this.setFirstRenderInputs = this.setFirstRenderInputs.bind(this);
        this.handleCargoItemQ = this.handleCargoItemQ.bind(this);
        this.handleCargoItemType = this.handleCargoItemType.bind(this);
    }

    handleCargoChange(event) {
        const { name, value } = event.target;
        this.setState({
            newCargoItem: { ...this.state.newCargoItem, [name]: value }
        });
    }

    setFirstRenderInputs(bool) {
        this.setState({firstRenderInputs: bool});
    }

    addNewCargo() {
        this.props.addCargoItem();
        this.setState({firstRenderInputs: true});
    }
    handleCargoItemQ(event) {
        const modifiedEvent = { target: event };
        this.props.handleDelta(modifiedEvent);
    }
    handleCargoItemType(event) {
        const index = event.name.split('-')[0];
        const modifiedEvent = {
            target: { name: event.name, value: event.key }
        };
        const newCargoItemTypes = this.state.cargoItemTypes;
        newCargoItemTypes[index] = event;
        this.setState({cargoItemTypes: newCargoItemTypes});
        this.props.handleDelta(modifiedEvent);
    }
    toggleDangerousGoods() {
        const event = {
            target: {
                name: 'dangerousGoods',
                value: !this.props.cargoItems[0].dangerousGoods
            }
        };
        // this.setState({ newCargoItem: { ...this.state.newCargoItem, dangerousGoods: !this.state.newCargoItem.dangerousGoods } });
        this.props.handleDelta(event);
    }
    deleteCargo(index) {
        this.props.deleteItem('cargoItems', index);
    }

    render() {
        const { cargoItems, handleDelta, theme } = this.props;
        const { cargoItemTypes } = this.state;
        const cargosAdded = [];
        const availableCargoItemTypes = this.props.availableCargoItemTypes ? (
            this.props.availableCargoItemTypes.map(cargoItemType => (
                {
                    label: cargoItemType.description,
                    key: cargoItemType.id,
                    dimension_x: cargoItemType.dimension_x,
                    dimension_y: cargoItemType.dimension_y
                }
            ))
        ) : [];
        const numberOptions = [];
        for (let i = 1; i <= 20; i++) {
            numberOptions.push({label: i, value: i});
        }
        const textStyle = {
            background: theme && theme.colors ? '-webkit-linear-gradient(left, ' + theme.colors.primary + ',' + theme.colors.secondary + ')' : 'black'
        };
        const generateSeparator = () => (
            <div key={v4()} className={`${styles.separator} flex-100`}>
                <hr/>
            </div>
        );
        if (cargoItems) {
            cargoItems.forEach((cont, i) => {
                if (i > 0) cargosAdded.push(generateSeparator());
                const tmpCont = (
                    <div
                        key={i}
                        className="layout-row flex-100 layout-wrap layout-align-start-center"
                    >
                        <div className="layout-row flex-90 layout-wrap layout-align-start-center" >
                            <div className="layout-row flex-100 layout-wrap layout-align-start-center" >
                                <div className="layout-row flex layout-wrap layout-align-start-center" >
                                    <p className={`${styles.input_label} flex-100`}> Gross Weight </p>
                                    <div className={`flex-95 layout-row ${styles.input_box}`}>
                                        <ValidatedInput
                                            className="flex-80"
                                            name={`${i}-payload_in_kg`}
                                            value={cont.payload_in_kg}
                                            type="number"
                                            onChange={handleDelta}
                                            firstRenderInputs={this.state.firstRenderInputs}
                                            setFirstRenderInputs={this.setFirstRenderInputs}
                                            nextStageAttempt={this.props.nextStageAttempt}
                                            validations={ {matchRegexp: /[^0]/} }
                                            validationErrors={ {matchRegexp: 'Must not be 0', isDefaultRequiredValue: 'Must not be blank'} }
                                            required
                                        />
                                        <div className="flex-20 layout-row layout-align-center-center">
                                            kg
                                        </div>
                                    </div>
                                </div>
                                <div className="layout-row flex layout-wrap layout-align-start-center" >
                                    <p className={`${styles.input_label} flex-100`}> Height </p>
                                    <div className={`flex-95 layout-row ${styles.input_box}`}>
                                        <ValidatedInput
                                            className="flex-80"
                                            name={`${i}-dimension_z`}
                                            value={cont.dimension_z}
                                            type="number"
                                            min="0"
                                            step="any"
                                            onChange={handleDelta}
                                            firstRenderInputs={this.state.firstRenderInputs}
                                            setFirstRenderInputs={this.setFirstRenderInputs}
                                            nextStageAttempt={this.props.nextStageAttempt}
                                            validations={ {matchRegexp: /[^0]/} }
                                            validationErrors={ {matchRegexp: 'Must not be 0', isDefaultRequiredValue: 'Must not be blank'} }
                                            required
                                        />
                                        <div className="flex-20 layout-row layout-align-center-center">
                                            cm
                                        </div>
                                    </div>
                                </div>
                                <div className="layout-row flex-50 layout-wrap layout-align-start-center" >
                                    <p className={`${styles.input_label} flex-100`}> Colli Type </p>
                                    <NamedSelect
                                        placeholder="Colli Type"
                                        className={styles.select_100}
                                        name={`${i}-colliType`}
                                        value={cargoItemTypes[i]}
                                        options={availableCargoItemTypes}
                                        onChange={this.handleCargoItemType}
                                    />
                                </div>
                            </div>
                            <div className="layout-row flex-100 layout-wrap layout-align-start-center" >
                                <div className="layout-row flex layout-wrap layout-align-start-center" >
                                    <p className={`${styles.input_label} flex-100`}> Length </p>
                                    <div className={`flex-95 layout-row ${styles.input_box}`}>
                                        <ValidatedInput
                                            className="flex-80"
                                            name={`${i}-dimension_x`}
                                            value={cont.dimension_x}
                                            type="number"
                                            min="0"
                                            step="any"
                                            onChange={handleDelta}
                                            firstRenderInputs={this.state.firstRenderInputs}
                                            setFirstRenderInputs={this.setFirstRenderInputs}
                                            nextStageAttempt={this.props.nextStageAttempt}
                                            validations={ {matchRegexp: /[^0]/} }
                                            validationErrors={ {matchRegexp: 'Must not be 0', isDefaultRequiredValue: 'Must not be blank'} }
                                            required
                                        />
                                        <div className="flex-20 layout-row layout-align-center-center">
                                            cm
                                        </div>
                                    </div>
                                </div>
                                <div className="layout-row flex layout-wrap layout-align-start-center" >
                                    <p className={`${styles.input_label} flex-100`}> Width </p>
                                    <div className={`flex-95 layout-row ${styles.input_box}`}>
                                        <ValidatedInput
                                            className="flex-80"
                                            name={`${i}-dimension_y`}
                                            value={cont.dimension_y}
                                            type="number"
                                            min="0"
                                            step="any"
                                            onChange={handleDelta}
                                            firstRenderInputs={this.state.firstRenderInputs}
                                            setFirstRenderInputs={this.setFirstRenderInputs}
                                            nextStageAttempt={this.props.nextStageAttempt}
                                            validations={ {matchRegexp: /[^0]/} }
                                            validationErrors={ {matchRegexp: 'Must not be 0', isDefaultRequiredValue: 'Must not be blank'} }
                                            required
                                        />
                                        <div className="flex-20 layout-row layout-align-center-center">
                                            cm
                                        </div>
                                    </div>
                                </div>
                                <div className="layout-row flex layout-wrap layout-align-start-center" >
                                    <p className={`${styles.input_label} flex-100`}> No. of Cargo Items </p>
                                    <NamedSelect
                                        placeholder={cont.quantity}
                                        className={styles.select}
                                        name={`${i}-quantity`}
                                        value={cont.quantity}
                                        options={numberOptions}
                                        onChange={this.handleCargoItemQ}
                                    />
                                </div>
                                <div className="layout-row flex layout-wrap layout-align-start-center" >
                                    <p className={`${styles.input_label} flex-100`}> Dangerous Goods </p>
                                    <Checkbox
                                        onChange={this.toggleDangerousGoods}
                                        checked={cont.dangerousGoods}
                                        theme={this.props.theme}
                                        size="34px"
                                    />
                                </div>
                            </div>
                        </div>
                        <div className="flex-10 layout-row layout-align-center-center">
                            <i className="fa fa-trash flex-none" onClick={() => this.deleteCargo(i)}></i>
                        </div>
                    </div>
                );
                cargosAdded.push(tmpCont);
            });
        }

        return (
            <div className="layout-row flex-100 layout-wrap layout-align-center-center" >
                <div className={`layout-row flex-none ${defs.content_width} layout-wrap layout-align-center-center section_padding`} >
                    <div className="layout-row flex-100 layout-wrap layout-align-start-center" >
                        { cargosAdded }
                        <div className="layout-row layout-align-start-center flex-100" >
                            <div className={`layout-row flex-none ${styles.add_unit} layout-wrap layout-align-center-center`} onClick={this.addNewCargo}>
                                <i className="fa fa-plus-square-o clip" style={textStyle}/>
                                <p> Add another </p>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

ShipmentCargoItems.propTypes = {
    theme: PropTypes.object,
    deleteItem: PropTypes.func,
    cargoItems: PropTypes.array,
    addCargoItem: PropTypes.func,
    handleDelta: PropTypes.func
};
