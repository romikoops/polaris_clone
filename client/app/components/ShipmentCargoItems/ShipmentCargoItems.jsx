import React, { Component } from 'react'
import { withNamespaces } from 'react-i18next'
import styles from './ShipmentCargoItems.scss'
import defs from '../../styles/default_classes.scss'
import QuantityInput from '../QuantityInput/QuantityInput'
import '../../styles/select-css-custom.scss'
import getInputs from './inputs'

class ShipmentCargoItems extends Component {
  constructor (props) {
    super(props)
    this.state = {
      cargoItemTypes: [],
      cargoItemInfoExpanded: [true]
    }
    this.addNewCargo = this.addNewCargo.bind(this)
    this.setFirstRenderInputs = this.setFirstRenderInputs.bind(this)
    this.handleCargoItemType = this.handleCargoItemType.bind(this)
  }

  componentWillMount () {
    const { cargoItems } = this.props
    const cargoItemTypes = cargoItems.map(cargoItem => (
      this.props.availableCargoItemTypes.find(cargoItemType => (
        +cargoItemType.key === +cargoItem.cargo_item_type_id
      ))
    ))
    const cargoItemInfoExpanded = cargoItems.map(() => true)
    this.setState({ cargoItemTypes, cargoItemInfoExpanded })
  }

  setFirstRenderInputs (bool) {
    this.setState({ firstRenderInputs: bool })
  }

  addNewCargo () {
    const { cargoItemInfoExpanded } = this.state
    cargoItemInfoExpanded.push(true)
    this.props.addCargoItem()
    this.setState({ firstRenderInputs: true, cargoItemInfoExpanded })
  }

  handleCargoItemType (event) {
    const index = event.name.split('-')[0]
    const modifiedEvent = {
      target: { name: `${index}-cargo_item_type_id`, value: event.key }
    }
    const newCargoItemTypes = this.state.cargoItemTypes
    newCargoItemTypes[index] = event
    this.setState({ cargoItemTypes: newCargoItemTypes })
    this.props.handleDelta(modifiedEvent, !event.key)

    if (!event.dimension_x) return

    const modifiedEventDimentionX = {
      target: { name: `${index}-dimension_x`, value: event.dimension_x }
    }
    const modifiedEventDimentionY = {
      target: { name: `${index}-dimension_y`, value: event.dimension_y }
    }
    this.props.handleDelta(modifiedEventDimentionX)
    this.props.handleDelta(modifiedEventDimentionY)
  }

  toggleCheckbox (value, e) {
    const artificialEvent = {
      target: { name: e.target.name, value }
    }
    this.props.handleDelta(artificialEvent)
  }

  toggleCargoItemInfoExpanded (i) {
    const { cargoItemInfoExpanded } = this.state
    cargoItemInfoExpanded[i] = !cargoItemInfoExpanded[i]
    this.setState({ cargoItemInfoExpanded })
  }

  deleteCargo (index) {
    const { cargoItemTypes } = this.state
    cargoItemTypes.splice(index, 1)
    this.setState({ cargoItemTypes })

    this.props.deleteItem('cargoItems', index)
  }

  render () {
    const {
      cargoItems,
      theme,
      toggleModal,
      nextStageAttempt,
      scope,
      handleDelta,
      maxDimensions,
      availableCargoItemTypes,
      availableMotsForRoute,
      t
    } = this.props
    const { cargoItemTypes, firstRenderInputs, cargoItemInfoExpanded } = this.state
    const cargosAdded = []
    const numberOptions = []
    for (let i = 1; i <= 20; i++) {
      numberOptions.push({ label: i, value: i })
    }
    const textStyle = {
      background:
        theme && theme.colors
          ? `-webkit-linear-gradient(left, ${theme.colors.primary},${theme.colors.secondary})`
          : 'black'
    }
    const generateCargoItem = (cargoItem, i) => {
      const inputs = getInputs.call(
        this,
        cargoItem,
        i,
        theme,
        cargoItemTypes,
        availableCargoItemTypes,
        numberOptions,
        firstRenderInputs,
        toggleModal,
        nextStageAttempt,
        scope,
        maxDimensions,
        availableMotsForRoute,
        t
      )

      return (
        <div
          key={i}
          name={`${i}-cargoItem`}
          className="layout-row flex-100 layout-wrap layout-align-stretch"
          style={{ position: 'relative', margin: '30px 0' }}
        >
          <div className={`flex-100 layout-align-start-center layout-row ${styles.cargo_unit_header}`}>
            <h3>{t('cargo:yourCargo')}</h3>
            {cargoItem ? (
              <div className={styles.delete_icon} onClick={() => this.deleteCargo(i)}>
                {t('common:delete')}
                <i className="fa fa-trash" />
              </div>
            ) : (
              ''
            )}
          </div>
          <div className={`flex-100 layout-row layout-wrap ${styles.cargo_unit_inputs}`}>
            <div className="flex-15 layout-row layout-align-center">
              <QuantityInput
                i={i}
                cargoItem={cargoItem}
                handleDelta={handleDelta}
                nextStageAttempt={nextStageAttempt}
              />
            </div>
            <div className={`${styles.cargo_item_box} ${styles.cargo_item_inputs} flex-85`}>
              <div style={{ position: 'relative' }}>
                <div
                  className={`layout-row flex-100 layout-wrap layout-align-start-center ${styles.padding_section}`}
                  style={{ marginBottom: '20px' }}
                >
                  {inputs.length}
                  {inputs.width}
                  {inputs.height}
                  <div className="flex-5" />
                  {scope.frontend_consolidation ? inputs.collectiveWeight : inputs.grossWeight}
                </div>
                <div className="flex-100 layout-row" />
                <div
                  className={`layout-row flex-100 layout-wrap layout-align-start-center ${styles.padding_section}`}
                  style={{ margin: '20px 0' }}
                >
                  {inputs.colliType}
                  {inputs.nonStackable}
                  {inputs.dangerousGoods}
                </div>
              </div>
              <div
                className={
                  `${styles.cargo_item_info} ` +
              `${cargoItemInfoExpanded[i] && styles.expanded} ` +
              'flex-100'
                }
              >
                <div
                  className={
                    `${styles.inner_cargo_item_info} layout-row flex-100 layout-wrap layout-align-start`
                  }
                >
                  <div className="flex-25 layout-wrap layout-row">
                    {inputs.totalVolume}
                    {inputs.chargeableVolume}
                  </div>
                  <div className={`${styles.padding_left} flex-25 layout-wrap layout-row`}>
                    {inputs.totalWeight}
                    {inputs.chargeableWeight}
                  </div>
                </div>
              </div>

            </div>

          </div>

        </div>
      )
    }

    if (cargoItems) {
      cargoItems.forEach((cargoItem, i) => {
        cargosAdded.push(generateCargoItem(cargoItem, i))
      })
    }

    return (
      <div className="layout-row flex-100 layout-wrap layout-align-center-center">
        <div
          className={
            `layout-row flex-none layout-wrap layout-align-center-center ${defs.content_width} `
          }
        >
          <div className="layout-row flex-100 layout-wrap layout-align-start-center">
            {cargosAdded}
          </div>

          <div className="layout-row flex-100 layout-wrap layout-align-start-center">
            <div className={`${styles.add_unit_wrapper} content_width`}>
              <div
                className={
                  `layout-row flex-none ${styles.add_unit} ` +
                  'layout-wrap layout-align-center-center'
                }
                onClick={this.addNewCargo}
              >
                <i className="fa fa-plus-square-o clip" style={textStyle} />
                <p> {t('shipment:addUnit')}</p>
              </div>
            </div>
          </div>
        </div>
        <style>
          {`
            .colli_type .Select-control {
              display: flex;
              position: relative;
              border: 1px solid #E0E0E0;
              box-shadow: none;
              background: transparent;
            }
            .colli_type .Select {
              box-shadow: none;
            }
            .colli_type .Select-value {
              background: transparent;
              box-shadow: none;
              border: 1px solid #E0E0E0;
            }
            .colli_type .Select-placeholder {
              background: transparent;
              box-shadow: none;
              padding-bottom: 5px;
              border: 1px solid #E0E0E0;
            }
            .colli_type .Select-clear-zone {
              position: absolute;
              right: 25px;
              top: 6px;
            }
            .colli_type .Select-arrow-zone {
              position: absolute;
              right: 0;
              top: 10px;
            }
          `}
        </style>
      </div>
    )
  }
}

ShipmentCargoItems.defaultProps = {
  theme: null,
  toggleModal: null,
  nextStageAttempt: false,
  cargoItems: [],
  availableCargoItemTypes: [],
  availableMotsForRoute: []
}

export default withNamespaces(['shipment', 'common', 'cargo', 'errors'])(ShipmentCargoItems)
