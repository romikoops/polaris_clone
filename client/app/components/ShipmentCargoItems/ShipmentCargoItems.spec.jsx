import * as React from 'react'
import { shallow } from 'enzyme'
import { theme, identity } from '../../mocks'

jest.mock('../TextHeading/TextHeading', () => {
  // eslint-disable-next-line react/prop-types
  const TextHeading = ({ children }) => <h2>{children}</h2>

  return {
    TextHeading
  }
})

jest.mock('../QuantityInput/QuantityInput', () => {
  const QuantityInput = props => <input {...props} />

  return {
    default: QuantityInput
  }
})
// eslint-disable-next-line import/first
import { ShipmentCargoItems } from './ShipmentCargoItems'

const propsBase = {
  theme,
  deleteItem: identity,
  cargoItems: [{
    description: 'FOO',
    key: 5,
    dimension_x: 13,
    dimension_y: 24,
    dangerous_goods: false,
    stackable: false
  }],
  availableCargoItemTypes: [{
    description: 'BAR',
    key: 7,
    dimension_x: 5,
    dimension_y: 6
  }],
  addCargoItem: identity,
  handleDelta: identity,
  toggleModal: identity,
  nextStageAttempt: false,
  scope: {
    modes_of_transport: {},
    dangerous_goods: false
  },
  maxDimensions: { x: 44, y: 33, z: 22 }
}

const createShallow = propsInput => shallow(<ShipmentCargoItems {...propsInput} />)

test.skip('shallow rendering', () => {
  expect(createShallow(propsBase)).toMatchSnapshot()
})
