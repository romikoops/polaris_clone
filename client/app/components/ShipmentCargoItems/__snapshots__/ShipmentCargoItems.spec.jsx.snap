// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow rendering 1`] = `
<div
  className="layout-row flex-100 layout-wrap layout-align-center-center"
>
  <div
    className="layout-row flex-none layout-wrap layout-align-center-center content_width "
  >
    <div
      className="layout-row flex-100 layout-wrap layout-align-start-center"
    >
      <div
        className="layout-row flex-100 layout-wrap layout-align-stretch"
        key="0"
        name="0-cargoItem"
        style={
          Object {
            "margin": "30px 0",
            "position": "relative",
          }
        }
      >
        <div
          className="flex-100 layout-align-start-center layout-row cargo_unit_header"
        >
          <h3>
            Cargo Group {{index}}
          </h3>
          <div
            className="delete_icon"
            onClick={[Function]}
          >
            Delete
            <i
              className="fa fa-trash"
            />
          </div>
        </div>
        <div
          className="flex-100 layout-row layout-wrap cargo_unit_inputs"
        >
          <div
            className="flex-15 layout-row layout-align-center"
          >
            <[object Object]
              cargoItem={
                Object {
                  "dangerous_goods": false,
                  "description": "FOO",
                  "dimension_x": 13,
                  "dimension_y": 24,
                  "key": 5,
                  "stackable": false,
                }
              }
              handleDelta={[Function]}
              i={0}
              nextStageAttempt={false}
            />
          </div>
          <div
            className="cargo_item_box cargo_item_inputs flex-85"
          >
            <div
              style={
                Object {
                  "position": "relative",
                }
              }
            >
              <div
                className="layout-row flex-100 layout-wrap layout-align-start-center padding_section"
                style={
                  Object {
                    "marginBottom": "20px",
                  }
                }
              >
                <div
                  className="layout-row flex-20 layout-wrap layout-align-start-center input_length"
                >
                  <h4>
                    Length
                  </h4>
                  <div
                    className="flex-90 layout-row input_box"
                    data-event="disable"
                    data-tip=""
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                  >
                    <img
                      alt="length"
                      border="0"
                      data-for="RANDOM_KEY_0"
                      data-tip="Length"
                    />
                    <ValidatedInput
                      errorStyles={
                        Object {
                          "fontSize": "10px",
                          "maxWidth": "200px",
                          "top": "32px",
                          "whiteSpace": "normal",
                        }
                      }
                      min="0"
                      name="0-dimension_x"
                      nextStageAttempt={false}
                      onChange={[Function]}
                      placeholder="0"
                      required={true}
                      setFirstRenderInputs={[Function]}
                      step="any"
                      type="number"
                      validationErrors={
                        Object {
                          "isDefaultRequiredValue": "Must be greater than 0",
                          "maxDimension": <p>
                            Maximum length is undefined
                            <br />
                            <span
                              className="emulate_link blue_link"
                              onClick={[Function]}
                            >
                              Request excess dimensions
                            </span>
                          </p>,
                          "nonNegative": "Must be greater than 0",
                        }
                      }
                      validations={
                        Object {
                          "maxDimension": [Function],
                          "nonNegative": [Function],
                        }
                      }
                      value={13}
                      wrapperClassName="flex-60"
                    />
                    <div
                      className="flex-25 layout-row layout-align-center-center"
                    >
                      cm
                    </div>
                  </div>
                </div>
                <div
                  className="layout-row flex-20 layout-wrap layout-align-start-center input_width"
                >
                  <h4>
                    Width
                  </h4>
                  <div
                    className="flex-90 layout-row input_box"
                    data-event="disable"
                    data-tip=""
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                  >
                    <img
                      alt="width"
                      border="0"
                      data-for="RANDOM_KEY_0"
                      data-tip="Width"
                    />
                    <ValidatedInput
                      errorStyles={
                        Object {
                          "fontSize": "10px",
                          "maxWidth": "200px",
                          "top": "32px",
                          "whiteSpace": "normal",
                        }
                      }
                      min="0"
                      name="0-dimension_y"
                      nextStageAttempt={false}
                      onChange={[Function]}
                      placeholder="0"
                      required={true}
                      setFirstRenderInputs={[Function]}
                      step="any"
                      type="number"
                      validationErrors={
                        Object {
                          "isDefaultRequiredValue": "Must be greater than 0",
                          "maxDimension": <p>
                            Maximum width is undefined
                            <br />
                            <span
                              className="emulate_link blue_link"
                              onClick={[Function]}
                            >
                              Request excess dimensions
                            </span>
                          </p>,
                          "nonNegative": "Must be greater than 0",
                        }
                      }
                      validations={
                        Object {
                          "maxDimension": [Function],
                          "nonNegative": [Function],
                        }
                      }
                      value={24}
                      wrapperClassName="flex-60"
                    />
                    <div
                      className="flex-25 layout-row layout-align-center-center"
                    >
                      cm
                    </div>
                  </div>
                </div>
                <div
                  className="layout-row flex-20 layout-wrap layout-align-start-center input_height"
                >
                  <h4>
                    Height
                  </h4>
                  <div
                    className="flex-90 layout-row input_box"
                    data-event="disable"
                    data-tip=""
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onMouseEnter={[Function]}
                  >
                    <img
                      alt="height"
                      border="0"
                      data-for="RANDOM_KEY_0"
                      data-tip="Height"
                    />
                    <ValidatedInput
                      errorStyles={
                        Object {
                          "fontSize": "10px",
                          "maxWidth": "200px",
                          "top": "32px",
                          "whiteSpace": "normal",
                        }
                      }
                      min="0"
                      name="0-dimension_z"
                      nextStageAttempt={false}
                      onChange={[Function]}
                      placeholder="0"
                      required={true}
                      setFirstRenderInputs={[Function]}
                      step="any"
                      type="number"
                      validationErrors={
                        Object {
                          "isDefaultRequiredValue": "Must be greater than 0",
                          "maxDimension": <p>
                            Maximum height is undefined
                            <br />
                            <span
                              className="emulate_link blue_link"
                              onClick={[Function]}
                            >
                              Request excess dimensions
                            </span>
                          </p>,
                          "nonNegative": "Must be greater than 0",
                        }
                      }
                      validations={
                        Object {
                          "maxDimension": [Function],
                          "nonNegative": [Function],
                        }
                      }
                      value=""
                      wrapperClassName="flex-60"
                    />
                    <div
                      className="flex-25 layout-row layout-align-center-center"
                    >
                      cm
                    </div>
                  </div>
                </div>
                <div
                  className="flex-5"
                />
                <div
                  className="layout-row flex-30 layout-wrap layout-align-start-center input_weight"
                >
                  <h4>
                    Gross Weight per item
                  </h4>
                  <div
                    className="flex-60 layout-row input_box"
                  >
                    <img
                      alt="weight"
                      border="0"
                      data-for="RANDOM_KEY_0"
                      data-tip="Gross Weight"
                    />
                    <Tooltip
                      color="#333"
                      icon="fa-info-circle"
                      text="payload_in_kg"
                      theme={Object {}}
                      toolText=""
                    />
                    <ValidatedInput
                      errorStyles={
                        Object {
                          "fontSize": "10px",
                          "maxWidth": "200px",
                          "top": "32px",
                          "whiteSpace": "normal",
                        }
                      }
                      name="0-payload_in_kg"
                      nextStageAttempt={false}
                      onChange={[Function]}
                      placeholder="0"
                      required={true}
                      setFirstRenderInputs={[Function]}
                      type="number"
                      validationErrors={
                        Object {
                          "isDefaultRequiredValue": "Must be greater than 0",
                          "maxDimension": <p>
                            Maximum weight is undefined
                            <br />
                            <span
                              className="emulate_link blue_link"
                              onClick={[Function]}
                            >
                              Request excess dimensions
                            </span>
                          </p>,
                          "nonNegative": "Must be greater than 0",
                        }
                      }
                      validations={
                        Object {
                          "maxDimension": [Function],
                          "nonNegative": [Function],
                        }
                      }
                      value=""
                      wrapperClassName="flex-90"
                    />
                    <div
                      className="flex-20 layout-row layout-align-center-center"
                    >
                      kg
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="flex-100 layout-row"
              />
              <div
                className="layout-row flex-100 layout-wrap layout-align-start-center padding_section"
                style={
                  Object {
                    "margin": "20px 0",
                  }
                }
              >
                <div
                  className="layout-row flex-40 layout-wrap layout-align-start-center colli_type"
                >
                  <div
                    style={
                      Object {
                        "width": "95%",
                      }
                    }
                  >
                    <NamedSelect
                      className="select_100"
                      inputProps={
                        Object {
                          "name": "0-colliType",
                        }
                      }
                      name="0-colliType"
                      onChange={[Function]}
                      options={
                        Array [
                          Object {
                            "description": "BAR",
                            "dimension_x": 5,
                            "dimension_y": 6,
                            "key": 7,
                          },
                        ]
                      }
                      placeholder="Select your colli type"
                      showErrors={false}
                    />
                  </div>
                </div>
                <div
                  className="layout-row flex layout-wrap layout-align-end-center cargo_unit_check"
                  onClick={[Function]}
                >
                  <Checkbox
                    checked={true}
                    className=""
                    disabled={true}
                    id="0-stackable"
                    name="0-stackable"
                    onChange={[Function]}
                    onClick={null}
                    size="15px"
                    theme={
                      Object {
                        "colors": Object {
                          "primary": "#333",
                          "secondary": "#fafafa",
                        },
                      }
                    }
                  />
                  <div
                    className="layout-row flex-65 layout-wrap layout-align-start-center"
                  >
                    <label
                      className="input_check flex-none pointy"
                      htmlFor="0-stackable"
                    >
                      <p>
                        Non Stackable
                      </p>
                    </label>
                    <Tooltip
                      color="#333"
                      icon="fa-info-circle"
                      text="non_stackable"
                      theme={Object {}}
                      toolText=""
                    />
                  </div>
                </div>
                <div
                  className="layout-row flex layout-wrap layout-align-start-center cargo_unit_check"
                  onClick={[Function]}
                >
                  <Checkbox
                    checked={false}
                    className=""
                    disabled={true}
                    id="0-dangerous_goods"
                    name="0-dangerous_goods"
                    onChange={[Function]}
                    onClick={null}
                    size="15px"
                    theme={
                      Object {
                        "colors": Object {
                          "primary": "#333",
                          "secondary": "#fafafa",
                        },
                      }
                    }
                  />
                  <div
                    className="layout-row flex-75 layout-wrap layout-align-start-center"
                  >
                    <label
                      className="input_check flex-none pointy"
                      htmlFor="0-dangerous_goods"
                    >
                      <p>
                        Dangerous Goods
                      </p>
                    </label>
                    <Tooltip
                      color="#333"
                      icon="fa-info-circle"
                      text="dangerous_goods"
                      theme={Object {}}
                      toolText=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="cargo_item_info expanded flex-100"
            >
              <div
                className="inner_cargo_item_info layout-row flex-100 layout-wrap layout-align-start"
              >
                <div
                  className="flex-25 layout-wrap layout-row"
                >
                  <div
                    className="flex-100 layout-row layout-wrap layout-align-start-center charge_volume"
                  >
                    <p
                      className="input_label flex-none"
                    >
                      Total
                       
                      Volume
                      :  
                      <span
                        className="input_value"
                      >
                         m
                        <sup
                          style={
                            Object {
                              "fontSize": "10px",
                              "height": "17px",
                              "marginLeft": "1px",
                            }
                          }
                        >
                          3
                        </sup>
                      </span>
                    </p>
                  </div>
                  <div
                    className="chargeable_weight layout-row flex-100 layout-wrap layout-align-start-center"
                  >
                    <div
                      className="layout-row flex-none layout-wrap layout-align-end-center"
                    >
                      <p
                        className="subchargeable flex-none"
                      >
                        Chargeable:
                      </p>
                    </div>
                    <div
                      className="chargeable_weight_values flex layout-row layout-align-start-center"
                    />
                  </div>
                </div>
                <div
                  className="padding_left flex-25 layout-wrap layout-row"
                >
                  <div
                    className="flex-100 layout-row layout-wrap layout-align-start-center charge_volume"
                  >
                    <p
                      className="input_label flex-none"
                    >
                      Total
                       
                      NO_TRANSLATION | key "common:weight"
                      :  
                      <span
                        className="input_value"
                      >
                         kg
                      </span>
                    </p>
                  </div>
                  <div
                    className="chargeable_weight layout-row flex-100 layout-wrap layout-align-start-center"
                  >
                    <div
                      className="layout-row flex-none layout-wrap layout-align-end-center"
                    >
                      <p
                        className="subchargeable flex-none"
                      >
                        Chargeable:
                      </p>
                    </div>
                    <div
                      className="chargeable_weight_values flex layout-row layout-align-start-center"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="layout-row flex-100 layout-wrap layout-align-start-center"
    >
      <div
        className="add_unit_wrapper content_width"
      >
        <div
          className="layout-row flex-none add_unit layout-wrap layout-align-center-center"
          onClick={[Function]}
        >
          <i
            className="fa fa-plus-square-o clip"
            style={
              Object {
                "background": "-webkit-linear-gradient(left, #333,#fafafa)",
              }
            }
          />
          <p>
             
            Add Cargo Group
          </p>
        </div>
      </div>
    </div>
  </div>
  <style>
    
            .colli_type .Select-control {
              display: flex;
              position: relative;
              border: 1px solid #E0E0E0;
              box-shadow: none;
              background: transparent;
            }
            .colli_type .Select {
              box-shadow: none;
            }
            .colli_type .Select-value {
              background: transparent;
              box-shadow: none;
              border: 1px solid #E0E0E0;
            }
            .colli_type .Select-placeholder {
              background: transparent;
              box-shadow: none;
              padding-bottom: 5px;
              border: 1px solid #E0E0E0;
            }
            .colli_type .Select-clear-zone {
              position: absolute;
              right: 25px;
              top: 6px;
            }
            .colli_type .Select-arrow-zone {
              position: absolute;
              right: 0;
              top: 10px;
            }
          
  </style>
</div>
`;
